crates/btcio/src/broadcaster/task.rs

```diff
@@ -35,147 +35,144 @@ pub async fn broadcaster_task(
-                        // So, set it to Unpublished
-                        L1TxStatus::Unpublished
-                    } else if info.confirmations >= params.rollup().l1_reorg_safe_depth.into() {
-                        L1TxStatus::Finalized {
-                            confirmations: info.confirmations,
-                        }
-                    } else {
-                        L1TxStatus::Confirmed {
-                            confirmations: info.confirmations,
-                        }
-                    }
-                }
-                Err(e) => {
-                    // If for some reasons tx is not found even if it was already
-                    // published/confirmed, set it to unpublished.
-                    if e.is_tx_not_found() {
-                        L1TxStatus::Unpublished
-                    } else {
-                        return Err(BroadcasterError::Other(e.to_string()));
-                    }
-                }
-            };
-            Ok(Some(new_status))
+            check_tx_confirmations(rpc_client, txentry, txid, params)
+                .await
+                .map(Some)
         }
-        L1TxStatus::Finalized { confirmations: _ } => Ok(None),
+        L1TxStatus::Finalized { .. } => Ok(None),
         L1TxStatus::InvalidInputs => Ok(None),
     }
 }
 
+async fn check_tx_confirmations(
+    rpc_client: &impl Wallet,
+    txentry: &L1TxEntry,
+    txid: Txid,
+    params: &Params,
+) -> BroadcasterResult<L1TxStatus> {
+    let txinfo_res = rpc_client.get_transaction(&txid).await;
+    debug!(?txentry.status, ?txinfo_res, ?txid, "check get transaction");
+
+    let reorg_safe_depth = params.rollup().l1_reorg_safe_depth.into();
+    match txinfo_res {
+        Ok(info) => match (info.confirmations, &txentry.status) {
+            // If it was published and still 0 confirmations, set it to published
+            (0, L1TxStatus::Published) => Ok(L1TxStatus::Published),
+
+            // If it was confirmed before and now it is 0, L1 reorged.
+            // So set it to Unpublished.
+            (0, _) => Ok(L1TxStatus::Unpublished),
+
+            (confirmations, _) if confirmations >= reorg_safe_depth => {
+                Ok(L1TxStatus::Finalized { confirmations })
+            }
+            (confirmations, _) => Ok(L1TxStatus::Confirmed { confirmations }),
+        },
+        Err(e) => {
+            // If for some reasons tx is not found even if it was already
+            // published/confirmed, set it to unpublished.
+            if e.is_tx_not_found() {
+                Ok(L1TxStatus::Unpublished)
+            } else {
+                Err(BroadcasterError::Other(e.to_string()))
+            }
+        }
+    }
+}
+
+async fn publish_tx(
+    rpc_client: &impl Broadcaster,
+    txentry: &L1TxEntry,
+    idx: u64,
+    txid: Txid,
+) -> BroadcasterResult<L1TxStatus> {
+    let tx = txentry.try_to_tx().expect("could not deserialize tx");
+    trace!(%idx, ?tx, "Publishing tx");
```

Comment:
Don't log the whole tx, it's excessive.  Just log the txid.  That's all we really care about and could find it in the database if needed.